src/main/java/net/minecraftforge/client/event/EntityViewRenderEvent.java

```diff
@@ -83,23 +90,67 @@ public void setDensity(float density)
     }
 
     /**
-     * Event that allows any feature to customize the rendering of fog.
+     * This event allows for customization of parameters related to fog rendering. The plane distances are based on the player's render distance.
+     *
+     * A FogMode of FOG_SKY is the skybox's fog.
+     * A FogMode of FOG_TERRAIN is what the fog induced by render distance uses. This works best for reducing the camera's visibility.
      */
-    @HasResult
-    public static class RenderFogEvent extends FogEvent
+    public static class RenderFogEvent extends EntityViewRenderEvent
     {
+        private final FogMode type;
         private final float farPlaneDistance;
+        private final float nearPlaneDistance;
+        private final FogShape fogShape;
 
+        @Deprecated //TODO: remove in 1.19
```

Comment:
This should be a `@Deprecated(forRemoval = true, since = "1.18.2")` annotation and a javadoc with a `@deprecated` tag saying that it will be removed in 1.19.